docs/geocode.rst

```diff
@@ -0,0 +1,142 @@
+Geocode
+=======
+
+The ``Geocode`` class provides geocoding using CARTO LDS.
+This process requires you to have a CARTO account with a geocoding provider and geocoding quota assigned, and its use will incurr in the expense of geocoding credits.
+In the case of accounts with soft geocoding limits, additional charges may apply if the monthly quota is exceeded.
+
+The ``geocode`` method provides the interface to geocoding; input data to be geocoded must be provided through a ``Dataset`` or ``DataFrame`` object as the first argument to this method.
+A second mandatory argument, ``street`` defines the name of the input data column that contains the street (postal address).
+
+Additional optional arguments can be used to define the ``city``, ``state`` and ``country``. These arguments can be used to either
+pass the name of a column that contains the corresponding attribute; e.g. ``city='column_name_of_the_city'``.
+Or a literal text within single quotes (useful if all rows of the dataset have the same attribute value), e.g. ``city="'London'"``.
+
+Another optional argument, ``metadata`` can define the name of a result column that will contain additional metadata about each gecododed row
+as a JSON string.
+
+The result of the ``geocode`` method is a tuple containing both a result Dataset
+(or a Dataframe, in case the input was a Dataframe) and a dictionary with general information about the process.
+
+Dry run
+-------
+
+To find out the number of quota credits that will be spent when geocoding a dataset pass a ``dry_run=True`` argument:
+
+.. code:: python
+
+    from cartoframes.data.services import Geocod
+    from cartoframes.data import Dataset
+    from cartoframes.auth import set_default_credentials
+
+    set_default_credentials(
+        username='YOUR_USERNAME',
+        api_key='YOUR_APIKEY'
+    )
+    gc = Geocode()
+
+    dataset = Dataset('YOUR_DATA')
+    _, info = gc.geocode(dataset, street='address', city='city', country="'Spain'", dry_run=True)
+    print(info.get('required_quota'))
+
+When ``dry_run`` is True no changes will be made to the data and no quota will be consumed.
+
+Geocoding Dataframes
+--------------------
+
+A Dataframe can be geocoded like this:
+
+.. code:: python
+
+    from cartoframes.data.services import Geocode
+    from cartoframes.data import Dataset
+    import pandas
```

Comment:
I know this is an annoying ask, but I think `import pandas` should be before all of the cartoframes imports. I see this in other examples below.